build.py

```diff
@@ -0,0 +1,71 @@
+# coding: utf-8
+
+import os
+import sys
+import subprocess
+import argparse
+import filecmp
+import tempfile
+
+
+class DockerOdooImages(object):
+
+    def __init__(self, folder, docker_image, git_check):
+        self._folder = folder
+        self._relpath = os.path.relpath(self._folder)
+        self._docker_image = docker_image
+        self._git_check = git_check
+        self._path = os.getcwd()
+        self._tempdir = os.path.join(tempfile.gettempdir(),
+                                     'docker-odoo-image')
+        if (not os.path.isdir(self._folder) or
+                not os.path.isfile(os.path.join(self._folder, 'Dockerfile'))):
+            raise Exception('No Dockerfile', 'The folder %s not containt'
+                            ' Dockerfile' % self._folder, ' or not exist')
+
+    def _dircmp(self, dcmp):
+        if dcmp.diff_files:
+            return True
+        for sub_dcmp in dcmp.subdirs.values():
+            return self._dircmp(sub_dcmp)
+        return False
+
+    def _has_been_changed(self):
+        return self._dircmp(filecmp.dircmp(self._folder,
+                                           os.path.join(self._tempdir,
+                                                        self._relpath)))
+
+    def build(self):
+        can_be_build = True
+        if self._git_check:
+            if not os.path.isdir(self._tempdir):
+                cmd = ["git", "clone",
```

Comment:
You don't need `clone` to check if a file has been changed.
You could use directly the remote like as [MQT-git_run](https://github.com/Vauxoo/maintainer-quality-tools/blob/233d67accf3da40f628ab64c49164ff6d9c9c6cb/travis/git_run.py#L24)